app/models/rule_result.rb

```diff
@@ -29,6 +29,15 @@ class RuleResult < ApplicationRecord
   PASSED = %w[pass].freeze
   FAIL = (SELECTED - PASSED).freeze
 
+  GROUP_BY_RESULT = Arel.sql(
+    <<-SQL
+    CASE WHEN result IN (#{sanitize_sql(RuleResult::FAIL.map { |rr| "'#{rr}'" }.join(','))}) THEN 'fail'
+         WHEN result IN (#{sanitize_sql(RuleResult::PASSED.map { |rr| "'#{rr}'" }.join(','))}) THEN 'pass'
```

Comment:
Can we NOT precompute it in SQL, but rather group it by result and then reshape it for a given API response. This increases the SQL query (readability) and generalizes the result into two states (when there are multiple states).